index.rst

```diff
@@ -1,61 +1,338 @@
+    This allows users with multiple institutional affiliations to use whichever authentication provider is most convenient for them.
+    [DMS-LSP-REQ-0024]
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+IDM-0005
+    Authenticated users must be able to view all federated identities attached to their account and revoke any of them.
 
-.. sectnum::
+IDM-0006
+    GitHub and Google must be supported as additional identity providers.
+    They may be added as additional identities to an existing account via IDM-0004, or be used by collaborators as described in IDM-0003.
+
+IDM-0007
+    Users with administrative access to the Science Platform (for example, the ability to impersonate other users) must authenticate with two authentication factors.
+    These will generally a password or some equivalent, and some additional security token bound to a sandboxed device (not a general-purpose laptop), such as an OTP authenticator on a mobile phone.
+    Since all authentication is via federated identity (IDM-0001), this must be implemented by requiring authentication from an external provider that, in turn, requires two-factor authentication.
+    GitHub or Google (IDM-0005), with appropriate configuration, would be suitable.
+
+IDM-0008
+    The authentication session generated by a web authentication must follow reasonable practices for secure cookie storage.
+    This means, at least, setting ``Secure`` and ``HttpOnly`` and using either opaque session tokens or cryptographically-signed data.
+
+IDM-0009
+    If a user primarily uses non-affiliation-providing identity providers such as Google and GitHub, there must be a mechanism to force the user to reauthenticate using an affiliation-providing identity provider to reconfirm their affiliation.
+    This shouldn't be done routinely, but may be necessary periodically to confirm that a user still has an appropriate affiliation.
+    One possible mechanism to implement this may be to add an expiration date for the user that is automatically extended when the user logs on with an appropriate identity provider.
+
+IDM-0010
+    Users may select their own account name.
+    This will be used as the username for Science Platform services and must not contain a ``@`` character (or any other character that causes issues for Science Platform services).
+    The account name must not match the name of any existing account.
+
+IDM-0011
+    Account names may not be reused except via (rare) administrative intervention to remove accounts that were erroneously created.
+
+IDM-0012
+    Users must be able to change their account name.
+    This change must propagate to all Science Platform services.
+    Science Platform services should therefore use a unique identifier rather than the account name (such as the numeric UID provided as part of the account metadata) wherever possible, and if not possible, must explicitly allow for account renaming.
```

Comment:
Yeah I'm not sure how the system would notify whoever it needs to that users have been renamed, but I guess there could be a central point or something.  Seems kind of tricky.  Using the UID seems better, although homedirs are what scare me.

Reply:
I've made it explicit in the requirements that any service that uses usernames has to have a handler for an account renaming event that does whatever is necessary (like rename home directories), and the identity management system has to be able to send those events.